### ISSUE TITLE
[Bug]: Thread creation failed: Resource temporarily unavailable in Querynode

### ISSUE THREAD

**Issue by linhgao** [milvus-io/milvus#18348]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: milvus-dev:2.1.0-20220719-66b6e0c7
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): 2.0.2
- OS(Ubuntu or CentOS): Ubuntu
- CPU/Memory: cpu: request 8/limit 16, mem: request 8Gi/limit 32Gi
- GPU: None
- Others:
```

### Current Behavior

When I test with 20 thread count, querynode pod restart.

### Expected Behavior

Should return search results normally

### Steps To Reproduce

```markdown
dim: 128
index type: HNSW
index params: {"M":64,"efConstruction":250}
metric_type: IP
collection_shard: 10
entity count: 670000

nq: 1
topK: 50
search_params: {"params": {"ef": 250}, "metric_type": "IP"}
thread count: 20
loop count: 10000
```

### Milvus Log

libgomp: Thread creation failed: Resource temporarily unavailable
Assert "metric_bimap.left.count(real_name)" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48
 => metric type not found: (IP)
Assert "metric_bimap.left.count(real_name)" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48
 => metric type not found: (IP)
[2022/07/20 02:35:38.140 +00:00] [WARN] [querynode/cgo_helper.go:57] ["Create Plan by expr failed, C Runtime Exception: [UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)\n"]
[2022/07/20 02:35:38.140 +00:00] [WARN] [querynode/task_scheduler.go:327] ["[UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)"]
[2022/07/20 02:35:38.140 +00:00] [WARN] [querynode/cgo_helper.go:57] ["Create Plan by expr failed, C Runtime Exception: [UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)\n"]
[2022/07/20 02:35:38.140 +00:00] [WARN] [querynode/task_scheduler.go:327] ["[UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)"]
Assert "metric_bimap.left.count(real_name)" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48
 => metric type not found: (IP)
[2022/07/20 02:35:38.141 +00:00] [WARN] [querynode/cgo_helper.go:57] ["Create Plan by expr failed, C Runtime Exception: [UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)\n"]
[2022/07/20 02:35:38.141 +00:00] [WARN] [querynode/task_scheduler.go:327] ["[UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)"]
Assert "metric_bimap.left.count(real_name)" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48
 => metric type not found: (IP)
[2022/07/20 02:35:38.141 +00:00] [WARN] [querynode/cgo_helper.go:57] ["Create Plan by expr failed, C Runtime Exception: [UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)\n"]
[2022/07/20 02:35:38.141 +00:00] [WARN] [querynode/task_scheduler.go:327] ["[UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)"]
Assert "metric_bimap.left.count(real_name)" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48
 => metric type not found: (IP)
[2022/07/20 02:35:38.141 +00:00] [WARN] [querynode/cgo_helper.go:57] ["Create Plan by expr failed, C Runtime Exception: [UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)\n"]
[2022/07/20 02:35:38.141 +00:00] [WARN] [querynode/task_scheduler.go:327] ["[UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)"]
[2022/07/20 02:35:38.141 +00:00] [ERROR] [grpcclient/client.go:233] ["ClientBase Call grpc first call get error"] [role=querynode] [error="err: rpc error: code = Canceled desc = context canceled\n, /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace\n/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:232 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).Call\n/go/src/github.com/milvus-io/milvus/internal/distributed/querynode/client/client.go:227 github.com/milvus-io/milvus/internal/distributed/querynode/client.(*Client).Search\n/go/src/github.com/milvus-io/milvus/internal/querynode/shard_cluster.go:780 github.com/milvus-io/milvus/internal/querynode.(*ShardCluster).Search.func2\n/usr/local/go/src/runtime/asm_amd64.s:1371 runtime.goexit\n"] [stack="github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).Call\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:233\ngithub.com/milvus-io/milvus/internal/distributed/querynode/client.(*Client).Search\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querynode/client/client.go:227\ngithub.com/milvus-io/milvus/internal/querynode.(*ShardCluster).Search.func2\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/shard_cluster.go:780"]
[2022/07/20 02:35:38.142 +00:00] [ERROR] [querynode/shard_cluster.go:797] ["stream operation failed: [UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)"] [stack="github.com/milvus-io/milvus/internal/querynode.(*ShardCluster).Search\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/shard_cluster.go:797\ngithub.com/milvus-io/milvus/internal/querynode.(*QueryNode).Search\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/impl.go:558\ngithub.com/milvus-io/milvus/internal/distributed/querynode.(*Server).Search\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querynode/service.go:298\ngithub.com/milvus-io/milvus/internal/proto/querypb._QueryNode_Search_Handler.func1\n\t/go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:4018\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/milvus-io/milvus/internal/proto/querypb._QueryNode_Search_Handler\n\t/go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:4020\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:1282\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:1616\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:921"]
[2022/07/20 02:35:38.142 +00:00] [WARN] [querynode/impl.go:560] ["search cluster failed"] [msgID=434707769025479720] [collectionID=434693517789626369] [error="stream operation failed: [UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)"]
Assert "metric_bimap.left.count(real_name)" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48
 => metric type not found: (IP)
[2022/07/20 02:35:38.142 +00:00] [WARN] [querynode/cgo_helper.go:57] ["Create Plan by expr failed, C Runtime Exception: [UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)\n"]
[2022/07/20 02:35:38.142 +00:00] [WARN] [querynode/task_scheduler.go:327] ["[UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)"]
Assert "metric_bimap.left.count(real_name)" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48
 => metric type not found: (IP)
[2022/07/20 02:35:38.142 +00:00] [WARN] [querynode/cgo_helper.go:57] ["Create Plan by expr failed, C Runtime Exception: [UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)\n"]
[2022/07/20 02:35:38.142 +00:00] [WARN] [querynode/task_scheduler.go:327] ["[UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)"]
Assert "metric_bimap.left.count(real_name)" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48
 => metric type not found: (IP)
[2022/07/20 02:35:38.142 +00:00] [WARN] [querynode/cgo_helper.go:57] ["Create Plan by expr failed, C Runtime Exception: [UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)\n"]
[2022/07/20 02:35:38.142 +00:00] [WARN] [querynode/task_scheduler.go:327] ["[UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)"]
Assert "metric_bimap.left.count(real_name)" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48
 => metric type not found: (IP)
[2022/07/20 02:35:38.143 +00:00] [WARN] [querynode/cgo_helper.go:57] ["Create Plan by expr failed, C Runtime Exception: [UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)\n"]
[2022/07/20 02:35:38.143 +00:00] [WARN] [querynode/task_scheduler.go:327] ["[UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)"]
Assert "metric_bimap.left.count(real_name)" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48
 => metric type not found: (IP)
[2022/07/20 02:35:38.143 +00:00] [WARN] [querynode/cgo_helper.go:57] ["Create Plan by expr failed, C Runtime Exception: [UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)\n"]
[2022/07/20 02:35:38.143 +00:00] [WARN] [querynode/task_scheduler.go:327] ["[UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP)"]
[2022/07/20 02:35:38.143 +00:00] [ERROR] [grpcclient/client.go:233] ["ClientBase Call grpc first call get error"] [role=querynode] [error="err: rpc error: code = Canceled desc = context canceled\n, /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace\n/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:232 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).Call\n/go/src/github.com/milvus-io/milvus/internal/distributed/querynode/client/client.go:227 github.com/milvus-io/milvus/internal/distributed/querynode/client.(*Client).Search\n/go/src/github.com/milvus-io/milvus/internal/querynode/shard_cluster.go:780 github.com/milvus-io/milvus/internal/querynode.(*ShardCluster).Search.func2\n/usr/local/go/src/runtime/asm_amd64.s:1371 runtime.goexit\n"] [stack="github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).Call\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:233\ngithub.com/milvus-io/milvus/internal/distributed/querynode/client.(*Client).Search\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querynode/client/client.go:227\ngithub.com/milvus-io/milvus/internal/querynode.(*ShardCluster).Search.func2\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/shard_cluster.go:780"]
[2022/07/20 02:35:38.143 +00:00] [ERROR] [querynode/shard_cluster.go:797] ["Search 13 failed, reason [UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP) err %!w(<nil>)"] [stack="github.com/milvus-io/milvus/internal/querynode.(*ShardCluster).Search\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/shard_cluster.go:797\ngithub.com/milvus-io/milvus/internal/querynode.(*QueryNode).Search\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/impl.go:558\ngithub.com/milvus-io/milvus/internal/distributed/querynode.(*Server).Search\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querynode/service.go:298\ngithub.com/milvus-io/milvus/internal/proto/querypb._QueryNode_Search_Handler.func1\n\t/go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:4018\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/milvus-io/milvus/internal/proto/querypb._QueryNode_Search_Handler\n\t/go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:4020\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:1282\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:1616\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:921"]
[2022/07/20 02:35:38.143 +00:00] [WARN] [querynode/impl.go:560] ["search cluster failed"] [msgID=434707769025479721] [collectionID=434693517789626369] [error="Search 13 failed, reason [UnexpectedError] Assert \"metric_bimap.left.count(real_name)\" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Types.cpp:48\n => metric type not found: (IP) err %!w(<nil>)"]

### Anything else?

After querynode restart, search failed.
![image](https://user-images.githubusercontent.com/102851605/179985779-9417a6c5-6522-4e16-83e8-ddf23c99db3d.png)

**Comment by yanliang567** [comment]
seems that the error pops up in knowhere?
@xige-16  could u please take a look first?
/assign @xige-16 
/unassign

**Comment by xiaofan-luan** [comment]
/assign @longjiquan 
thought long is working on it

**Comment by xiaofan-luan** [comment]
should be fixed on 2.1.1, with the limitation

**Comment by yanliang567** [comment]
@linhgao any chance to verify fix in v2.1.1, which is available soon

**Comment by xiaofan-luan** [comment]
should be fixed by #18542
